### PR TITLE
NNVP-44 IRES integration

### PR DESCRIPTION
**Pull request by eryk7b**

### CODE REVIEW

**Comment by laurent-theglue** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-32** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-32.cluster.development.nnvp.thegluesandbox.com)

File: src/test/java/com/theglue/nnvp/service/IRESServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@ActiveProfiles("test")
+public class IRESServiceTest {
+
+  @Autowired
+  private IRESService service;
+
+  @Test
+  public void testIRESInformationFetching() {
```
Comment (mateusztheglue) [inline src/test/java/com/theglue/nnvp/service/IRESServiceTest.java]: not sure what I think about unit test that requires another service to be accesible
Probably we should mock the IRES endopont response in the unit test.

File: src/test/java/com/theglue/nnvp/service/IRESServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    );
+    IRESResponse iresResponse1 = service.getByLicensePlate("1234567");
+    Assert.assertTrue(iresResponse1.getHeader()
+        .getStatus()
+        .stream()
+        .anyMatch(iresResponseHeaderStatus -> iresResponseHeaderStatus.getCode()
+            .getName()
+            .equals(ResponseHeaderStatusCodeName.PLATE_NOT_FOUND)));
+  }
+
```
Comment (mateusztheglue) [inline src/test/java/com/theglue/nnvp/service/IRESServiceTest.java]: would be nice to also test case with failing @Pattern validation

File: src/main/java/com/theglue/nnvp/service/IRESService.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    return getByLicensePlate(licensePlate, null);
+  }
+
+  public IRESResponse getByLicensePlate(String licensePlate, Map<String, String> parameters) {
+    String endpoint = getPlateEndpoint(licensePlate);
+    return getIRESResponse(endpoint, parameters);
+  }
+
+  private IRESResponse getIRESResponse(String endpoint, Map<String, String> parameters) {
+    ResponseEntity<IRESResponse> response = MapUtils.isNotEmpty(parameters)
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/service/IRESService.java]: handle timeout / error response from ires

**Pull request closed without merging**